lib/github_service.rb

```diff
@@ -113,6 +113,21 @@ def refresh_assignees(fq_name)
       assignees_cache.delete(fq_name)
     end
 
+    def username_lookup(username)
+      if username_lookup_cache.key?(username)
+        username_lookup_cache[username]
```

Comment:
Why the double lookup?  That is the `||=` below handles this case, so the if/else is not needed.

Reply:
No, it won't work with `nil`, because it will keep checking it.  `nil` in this case is used to denote that it is an invalid username.